vendor/models/offer.py

```diff
@@ -111,13 +112,9 @@ def get_best_currency(self, currency=DEFAULT_CURRENCY):
         """
         Gets best currency for prodcuts available in this offer
         """
-        supported_currencies = []
-        for product in self.products.all():
-            for supported_currency in product.meta['msrp'].keys():
-                if supported_currency != 'default':
-                    supported_currencies.append(supported_currency)
+        available_currencies = set(settings.AVAILABLE_CURRENCIES.keys()).intersection( *(set(product.meta['msrp'].keys()) for product in self.products.all() ) )
```

Comment:
Since this is the third location of nearly the same code, might be good idea to make it into a single function that can be shared for consistency.

Reply:
added the util function is_currency_available